### PR TITLE
[CBRD-23753] Fix destroying Internal JDBC resources is not working and unexpected behavior

### PR DESCRIPTION
**Pull request by hgryoo**
http://jira.cubrid.org/browse/CBRD-23753

- Disabled ssl_client in the destroying state of Java SP processing
- Added defensive code
    - Avoid caused unexpected NullPointerException in UStatementHandlerCache
    - Added checkReconnect() in UServerSideConnection.disconnect(). In some cases, output buffer (outBuffer) is not allocated and NullPointerException is caused.
    - Added REQ_CODE_END not to call libcas_main() when InternalJDBCConnection was not used in the stored procedure.

### CODE REVIEW

File: src/jsp/jsp_cl.c [hunk 1]
```diff
@@ -2174,22 +2174,42 @@ jsp_send_destroy_request_all ()
...
+      bool mode = ssl_client;
+      ssl_client = false;
+      tran_begin_libcas_function ();
+      libcas_main (sockfd);	/* jdbc call */
+      tran_end_libcas_function ();
+      ssl_client = mode;
+    }
+  else
+    {
+      /* end */
```
Comment (byungwook-kim) [inline src/jsp/jsp_cl.c:2209]: no necessary for else ?

Comment (hgryoo) [reply src/jsp/jsp_cl.c:2209]: If ServerSideConnection is not created in a stored procedure, entering into the main of CAS protocol, for closing cached UStatements, is not required. So ExecuteThread will send REQ_CODE_END to the CAS. And CAS will get out of the destroying state.

File: src/jsp/jsp_cl.c [hunk 2]
```diff
@@ -2174,22 +2174,42 @@ jsp_send_destroy_request_all ()
...
-  tran_begin_libcas_function ();
-  res = libcas_main (sockfd);	/* jdbc call to destroy resources */
-  tran_end_libcas_function ();
+  /* read request code */
+  int code;
+  nbytes = jsp_readn (sockfd, (char *) &code, (int) sizeof (int));
+  if (nbytes != (int) sizeof (int))
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_SP_NETWORK_ERROR, 1, nbytes);
+      return ER_SP_NETWORK_ERROR;
```
Comment (byungwook-kim) [inline src/jsp/jsp_cl.c:2194]: how about change to "return er_errid ()" as like prior jsp_writen error processing.

Comment (hgryoo) [reply src/jsp/jsp_cl.c:2194]: Ok, I'll fix it.

**Review by byungwook-kim — APPROVED** [review]

**Review by shparkcubrid — APPROVED** [review]

**Pull request merged**